docs/dg/dev/guidelines/performance-guidelines/external-http-requests.md

```diff
@@ -69,6 +104,70 @@ $promises = [
 $responses = Promise\Utils::unwrap($promises);
 ```
 
+### Asynchronous AJAX calls to avoid blocking page rendering
+
+When frontend pages depend on data from backend-gateway or other remote resources, avoid blocking the initial page render. Synchronous calls to external services can significantly delay the Time to First Byte (TTFB) and degrade user experience.
+
+#### The problem with blocking calls
+
+If a page waits for external data before rendering, users see a blank screen or loading spinner for the entire request duration. This is particularly problematic when:
+- External services are slow or temporarily unavailable
+- Multiple external calls are required sequentially
+- Network latency adds up across calls
+
+#### Solution: Asynchronous loading with proper UI design
+
+Instead of blocking page rendering, design your UI to load data asynchronously:
+
+1. **Render the page skeleton first**: Display the page structure, navigation, and static content immediately.
+2. **Use AJAX for dynamic data**: Fetch external data asynchronously after the page loads.
+3. **Provide visual feedback**: Show loading indicators, placeholders, or skeleton screens while data is being fetched.
+4. **Handle errors gracefully**: Display user-friendly error messages if external calls fail.
+
+#### Implementation example
+
+**Bad approach** - Blocking render:
+
+```php
+// Controller waits for external data before rendering
+$externalData = $this->externalClient->fetchData(); // Blocks for 2-3 seconds
+return $this->view(['data' => $externalData]);
+```
+
+**Good approach** - Asynchronous loading:
+
+```php
+// Controller renders immediately with placeholder
+return $this->view(['dataEndpoint' => '/api/external-data']);
+```
+
+```javascript
+// Frontend fetches data asynchronously
+fetch('/api/external-data')
```

Comment:
it looks to me as suggesting 3rd party call directly here, so I would suggest to mention this is the Spryker endpoint, which makes secure call to 3rd party and returns data.